### PR TITLE
Stop using deprecated React lifecycle hooks

### LINKED ISSUE #373: Use React strict mode
https://reactjs.org/docs/strict-mode.html

This will also let us remove this: https://github.com/hyperdexapp/hyperdex/blob/60d9feecda1449222ac914f92e247b6e2cf54957/package.json#L102

Assigning to @kevva since he has the most React experience.

### PR DESCRIPTION
**Pull request by kevva**
We have some third party components that causes some noise when using strict mode, but our components should be okay at least.

Fixes #373.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Let's not enable the Strict mode itself for now. It's very noisy. Just add a `TODO` comment about it to enable it before upgrading to React 17. The main offenders are `react-select` and `recharts`, which we can't easily improve, so better to just let the community them when React actually removes those methods.

Also @kevva is fixing `styled-jsx`: https://github.com/zeit/styled-jsx/pull/457

**Comment by sindresorhus** [comment]
I got this error when trying to withdraw some DOGE in the Withdraw modal:

```
Warning: Modal.getDerivedStateFromProps(): A valid state object (or null) must be returned. You have returned undefined.
```

**Comment by kevva** [comment]
> I got this error when trying to withdraw some DOGE in the Withdraw modal:

Didn't take into account when `state` hasn't changed in the `Modal` component.

**Comment by sindresorhus** [comment]
@kevva The modal doesn't close anymore after a successful withdrawal.

File: app/renderer/components/Modal.js [hunk 1]
```diff
@@ -11,20 +11,32 @@ class Modal extends React.Component {
 		delay: 400,
 	};
 
+	static getDerivedStateFromProps(props, state) {
+		if (!state.open && props.open) {
```
Comment (kevva) [inline app/renderer/components/Modal.js]: Lol, `state.open` should be `state.isOpen`.

**Comment by kevva** [comment]
@sindresorhus, fixed now. Besides a typo (`state.open` instead of `state.isOpen`), we also need to handle when `isOpen` is still `true` but the modal is closing.

**Comment by sindresorhus** [comment]
Works perfectly now :)

**Pull request merged**